src/Base/Exception.h

```diff
@@ -33,6 +33,13 @@
 #include "FileInfo.h"
 #include "BaseClass.h"
 
+/* MACROS FOR THROWING EXCEPTIONS */
+
+#define THROW(exception) throw Base::exception(__FILE__,__LINE__,__PRETTY_FUNCTION__);
+#define THROWM(exception, message) throw Base::exception(message,__FILE__,__LINE__,__PRETTY_FUNCTION__);

```

Comment:
One of the main ideas of doing this is that the message is separated from the debugging information (as to allow for translations of the message in the UI).

Once I separate message from debugging information, I need a constructor that takes (at least) two parameters. Once I am there, I think it makes sense to separate the debugging information into three variables. I have the same number of constructors, only they take 3 parameters instead of 1 (constructor without message), of 4 instead 2 (constructor with message).

As for typing, I think it is one of those corners where c++ forces you to copy/paste, as you have to define the constructors for each class. I weighted using the same syntax DeepSOIC uses in Exception containers (a macro to define full exception classes). However this would mean:
1. I have to change the whole file, which includes a lot of work by others. I am not that comfortable with that.
2. It makes the exception declarations way less clear.

Then I weighted in using a macro for inserting the new constructors (not the full class). As they are cases where there is a message and others where there is no message and combinations thereof, I would have to write almost the same amount of lines (with the exception that code may be easier to maintain, still less clear). 

I do not have a strong opinion on whether macros should be used to avoid a more verbose definition or a more verbose definition is preferred to avoid macro introduced lack of clarity. I can accept either.